### PR TITLE
Feature/studentlist

### PR DESCRIPTION
**Pull request by brauliodiez**
Includes training list + student list.

Training list has already been reviewed (we have made this interim step because it needes some from student list and was not in master).

Student list, there are things to fix (e.g. feedback provided in training list to apply in student).

Please could you go with this review, it can become a showstopper, we need to review it in order to move to master.

### CODE REVIEW

File: src/reducers/spec/adminStudent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Act
+    const newState = adminStudentReducer(originalState, actionResult);
+
+    // Assert
+    expect(newState.studentSummaryList).not.to.be.undefined;
+    expect(newState.studentSummaryList.length).to.be.equal(2);
+    expect(newState.studentSummaryList[0].id).to.be.equal(2);
+    expect(newState.studentSummaryList[1].id).to.be.equal(3);
+  })
```
Comment (franmolmedo) [inline src/reducers/spec/adminStudent.spec.ts]: It is the same with other expects... I think we can improve then. In this example we can use only the array comparition.

Comment (brauliodiez) [inline src/reducers/spec/adminStudent.spec.ts]: Good points, Will do

File: src/rest-api/spec/training.spec.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+      trainingApi.setMockDataSeed(trainingList);
+
+      const summaryListPromise = trainingApi.getSummaryTrainingList();
+
+      summaryListPromise.then((summaryList) => {
+        // Assert
+        expect(summaryList).not.to.be.undefined;
+        expect(summaryList.length).to.be.equal(trainingList.length);
+        expect(summaryList).to.eql(expectedSummaryList);
```
Comment (brauliodiez) [inline src/rest-api/spec/training.spec.ts]: Agree, maybe it was just natural flow testing, but worth to refactor it, cleaner to understand and maintain

**Comment by atobajas** [comment]
Proposal: delete all Interfaces without elements and change to stateless the components that not requiered statefull.

File: src/rest-api/spec/training.spec.ts (hunk 2 above)
Comment (rumbado) [inline src/rest-api/spec/training.spec.ts]: My two cents,

I like that breaking on assertions, if any of those assertions fails in the future you have more information of what is the problem just knowing at which one breaks

Comment (brauliodiez) [inline src/rest-api/spec/training.spec.ts]: Update done, will push in some minutes.

Comment (brauliodiez) [inline src/rest-api/spec/training.spec.ts]: I see both points, @franmolmedo strength is... less code easier to read and maintain, @rumbado take, incremental asserts easier to pinpoint when you have an issue.

**Pull request merged**